packages/discord.js/src/structures/BaseInvite.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+const { RouteBases } = require('discord-api-types/v10');
+const { Base } = require('./Base.js');
+
+/**
+ * The base invite class.
+ * @extends {Base}
+ * @abstract
+ */
+class BaseInvite extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The type of this invite
+     * @type {InviteType}
+     */
+    this.type = data.type;
+
+    /**
+     * The invite code.
+     * @type {string}
+     */
+    this.code = data.code;
+
+    this._patch(data);
+  }
+
+  _patch(data, guild) {
+    if ('inviter_id' in data) {
+      /**
+       * The id of the user that created this invite.
+       * @type {?Snowflake}
+       */
+      this.inviterId = data.inviter_id;
+    } else {
+      this.inviterId ??= null;
+    }
+
+    if ('inviter' in data) {
+      this.client.users._add(data.inviter);
+      this.inviterId ??= data.inviter.id;
+    }
+
+    if ('max_age' in data) {
+      /**
+       * The maximum age of the invite in seconds. `0` for no expiry.
+       * @type {?number}
+       */
+      this.maxAge = data.max_age;
+    } else {
+      this.maxAge ??= null;
+    }
+
+    if ('created_at' in data) {
+      /**
+       * The timestamp this invite was created at.
+       * @type {?number}
+       */
+      this.createdTimestamp = Date.parse(data.created_at);
+    } else {
+      this.createdTimestamp ??= null;
+    }
+
+    if ('expires_at' in data) {
+      this._expiresTimestamp = data.expires_at && Date.parse(data.expires_at);
+    } else {
+      this._expiresTimestamp ??= null;
+    }
+
+    if ('channel_id' in data) {
+      /**
+       * The id of the channel this invite is for.
+       * @type {?Snowflake}
+       */
+      this.channelId = data.channel_id;
+    }
+
+    if ('channel' in data) {
+      /**
+       * The channel this invite is for.
+       * @type {?BaseChannel}
+       */
+      this.channel = this.client.channels._add(data.channel, guild, { cache: false });
```

Comment:
```suggestion
      this.channel = this.client.channels._add(data.channel, guild, { cache: false }) ?? this.client.channels.resolve(this.channelId);
```

If `data.channel` is null but `data.channel_id` exists this would result in `this.channel` being null although it might exist in cache. This happens in `inviteDelete` event for example.

Reply:
Since the documentation does not state the channel object is emitted, I have done this.